internal/abi/fastly/httpcache_guest.go

```diff
@@ -0,0 +1,1107 @@
+	b prim.Pointer[httpIsSensitive],
+) FastlyStatus
+
+func HTTPCacheGetSensitiveData(h *HTTPCacheHandle) (bool, error) {
+	var b httpIsSensitive
+
+	if err := fastlyHTTPCacheGetSensitiveData(
+		h.h,
+		prim.ToPointer(&b),
+	).toError(); err != nil {
+		return false, err
+	}
+
+	return b != 0, nil
+}
+
+// witx:
+//
+//    ;;; Get the surrogate keys of the found response, returning the `$none` error if there was no
+//    ;;; found response.
+//    ;;;
+//    ;;; The output is a list of surrogate keys separated by spaces.
+//    ;;;
+//    ;;; If the guest-provided output parameter is not long enough to contain the full list of
+//    ;;; surrogate keys, the required size is written by the host to `nwritten_out` and the `$buflen`
+//    ;;; error is returned.
+//    (@interface func (export "get_surrogate_keys")
+//        (param $handle $http_cache_handle)
+//        (param $surrogate_keys_out_ptr (@witx pointer u8))
+//        (param $surrogate_keys_out_len (@witx usize))
+//        (param $nwritten_out (@witx pointer (@witx usize)))
+//        (result $err (expected (error $fastly_status)))
+//    )
+//
+
+//go:wasmimport fastly_http_cache get_surrogate_keys
+//go:noescape
+func fastlyHTTPCacheGetSurrogateKeys(
+	h httpCacheHandle,
+	buf prim.Pointer[prim.U8],
+	bufLen prim.Usize,
+	nwritten prim.Pointer[prim.Usize],
+) FastlyStatus
+
+func HTTPCacheGetSurrogateKeys(h *HTTPCacheHandle) (string, error) {
+	n := DefaultMediumBufLen
+
+	for {
+		buf := prim.NewWriteBuffer(n) // Longest (unknown)
+		status := fastlyHTTPCacheGetSurrogateKeys(
+			h.h,
+			prim.ToPointer(buf.U8Pointer()),
+			buf.Cap(),
+			prim.ToPointer(buf.NPointer()),
+		)
+		if status == FastlyStatusBufLen && buf.NValue() > 0 {
+			n = int(buf.NValue())
+			continue
+		}
+		if err := status.toError(); err != nil {
+			return "", err
+		}
+		return string(buf.AsBytes()), nil
+	}
+}
+
+// witx:
+//
+//    ;;; Get the vary rule of the found response, returning the `$none` error if there was no found
+//    ;;; response.
+//    ;;;
+//    ;;; The output is a list of header names separated by spaces.
+//    ;;;
+//    ;;; If the guest-provided output parameter is not long enough to contain the full list of
+//    ;;; surrogate keys, the required size is written by the host to `nwritten_out` and the `$buflen`
+//    ;;; error is returned.
+//    (@interface func (export "get_vary_rule")
+//        (param $handle $http_cache_handle)
+//        (param $vary_rule_out_ptr (@witx pointer u8))
+//        (param $vary_rule_out_len (@witx usize))
+//        (param $nwritten_out (@witx pointer (@witx usize)))
+//        (result $err (expected (error $fastly_status)))
+//    )
+//
+
+//go:wasmimport fastly_http_cache get_vary_rule
+//go:noescape
+func fastlyHTTPCacheGetVaryRule(
+	h httpCacheHandle,
+	buf prim.Pointer[prim.U8],
+	bufLen prim.Usize,
+	nwritten prim.Pointer[prim.Usize],
+) FastlyStatus
+
+func HTTPCacheGetVaryRule(h *HTTPCacheHandle) (string, error) {
+	n := DefaultMediumBufLen
```

Comment:
I bet this could be `DefaultSmallBufLen` since it will normally contain a list of a few header names or `*` for the degens.

Reply:
7c135d4